app/admin/page.tsx

```diff
@@ -0,0 +1,22 @@
+'use client'
+import { useEffect } from "react";
+import { AuthContent, useAuthContext } from "../context/AuthContext";
+import { useRouter } from "next/navigation";
+
+function Page() {
+    const { user }: AuthContent = useAuthContext();
+    const router = useRouter();
+
+    useEffect(() => {
```

Comment:
I think it is best to explicit void return types

```suggestion
    useEffect((): void => {
```